api/singapore/v1alpha1/registeredcluster_types.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Red Hat
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=="ManagedClusterJoined")].status`,name="Joined",type=string
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=="ManagedClusterConditionAvailable")].status`,name="Available",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// RegisteredClusterSpec defines the desired state of RegisteredCluster
+type RegisteredClusterSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+}
+
+// RegisteredClusterStatus defines the observed state of RegisteredCluster
+type RegisteredClusterStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//ImportCommandRef is reference to configmap containing import command.
+	ImportCommandRef importCommandRef `json:"importCommandRef"`
```

Comment:
In my head this was a `LocalObjectReference` versus just a string. Thoughts?

Reply:
yes we can use `LocalObjectReference`.